x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -161,7 +168,8 @@ export const useFetchIndex = (
 
                 previousIndexesName.current = response.indicesExist;
                 setLoading(false);
-
+                // TODO: Steph/sourcerer all the below formatting should be happening serverside
```

Comment:
talking about that, I. think we can simplify a lot of thing here. I was just checking the type of `DataViewSpec` and now the fields attributes is an object and it seems that the key is the field name

Reply:
`DataViewSpec['fields']` differs slightly from `IndexField` so I'm not sure we can just replace it
```
export interface IndexField {
  category: string; // missing from DataViewSpec
  example?: Maybe<string | number>; // missing from DataViewSpec
  indexes: Array<Maybe<string>>; // missing from DataViewSpec
  name: string; // exact same in DataViewSpec
  type: string; // exact same in DataViewSpec
  searchable: boolean; // exact same in DataViewSpec
  aggregatable: boolean; // ? not required in DataViewSpec
  subType?: IFieldSubType; // exact same in DataViewSpec
  description?: Maybe<string>; // missing from DataViewSpec
  format?: Maybe<string>; // DIFFERENT! SerializedFieldFormat
  esTypes?: string[]; // ? not required in DataViewSpec
  subType?: IFieldSubType; // exact same in DataViewSpec
  readFromDocValues: boolean; // ? not required in DataViewSpec
}
```

Reply:
let's investigate in a follow up PR